gtsam/hybrid/HybridNonlinearFactor.h

```diff
@@ -74,6 +74,13 @@ class GTSAM_EXPORT HybridNonlinearFactor : public HybridFactor {
   /// Decision tree of Gaussian factors indexed by discrete keys.
   Factors factors_;
 
+  /// HybridFactor method implementation. Should not be used.
```

Comment:
Hmmm. So then why does it exist in the base class. red flag. Should maybe only exist in Gaussian branch.

Reply:
This is primarily because of HybridConditional.

The class hierarchy is
- HybridFactor
  |- HybridConditional
  |- HybridGaussianFactor
    |- HybridGaussianConditional

HybridConditional acts as a common container class for GaussianConditional, DiscreteConditional and HybridGaussianConditional. By defining errorTree in HybridFactor, I can just check for HybridFactor. Otherwise, I'll have to check for HybridConditional and HybridGaussianFactor separately which I felt defeated the purpose of this base class method.

Reply:
So, this is a different hierarchy than the non-hybrid case then:
```
class GTSAM_EXPORT GaussianConditional :
    public JacobianFactor,
    public Conditional<JacobianFactor, GaussianConditional>
```

Reply:
Sorry there was a rendering error in my previous comment.
It is the same hierarchy as GaussianConditional since HybridGaussianConditional inherits from HybridGaussianFactor, however HybridConditional complicates things (or at least makes it messier).

Reply:
What would break if you only have errorTree in HybridGaussianFactor? It would be available in HGC. All done?

Reply:
Yes, but then we'd need a separate check for HybridConditional (which is different from HybridGaussianConditional).

I think I'm okay with that. The current scheme just seemed tidier (except for HybridNonlinearFactor which isn't tidy).

Reply:
Why is a check needed? We would never call errorTree on HybridConditional, as it only makes sense for Gaussian variants, right?

Reply:
A HybridConditional can have a GaussianConditional or a HybridGaussianConditional as its underlying conditional. 

Remember that HybridConditional is a container based on type erasure and not a base class for HybridGaussianConditional.

Reply:
Right. But then why would we need an errorTree function for HybridConditional. If it can be either, then a method providing VectorValues does not make sense to me. Where would that be used? Is it used???

Reply:
It is used in an incremental sense, when we add the conditionals from a HybridBayesNet to the HybridGaussianFactorGraph, the factor graph now has `HybridGaussianFactor` (and potential `HybridGaussianConditional`s) and `HybridConditional`s as well.

The alternative could be to unwrap the conditionals from the `HybridConditional` before adding it to the graph.

I am open to suggestions about this since this was nontrivial for me. I think checking for `HybridGaussianFactor` and `HybridConditional` separately makes sense.

Reply:
Oh and another case: if we go from a `HybridBayesNet` to a `HybridGaussianFactorGraph`, then all the factors are added as `HybridConditional`s which are represented as `HybridFactor`s. This causes `DiscreteFactor`s to be considered in the errorTree computations as well. :(